Aggregation $ifNull is causing ClassCastException

I'm using the following code to project **true** whenever the field (field_to_check) **is not null**, otherwise false:
```java
BasicDBObject project = new BasicDBObject();
project.append("is_field_defined", new BasicDBObject()
	.append("$cond", new BasicDBObject()
		.append("if", new BasicDBObject("$ifNull", Arrays.asList("$field_to_check", false)))
		.append("then", true)
		.append("else", false)
		)
);
```

This works well with MongoDB, but is causing an exception with FongoDB in my unittests:
```
Caused by: java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String
	at com.github.fakemongo.impl.aggregation.Project$ProjectedIfNull.<init>(Project.java:231)
	... 64 more
```

The problem seems to be in ```com.github.fakemongo.impl.aggregation.Project.ProjectedIfNull``` where the expected ```valueIfNull``` must be a String:

```java
  static class ProjectedIfNull extends ProjectedAbstract<ProjectedIfNull> {
    public static final String KEYWORD = "$ifNull";

    private final String field;
    private final String valueIfNull;

    public ProjectedIfNull(String destName, DBCollection coll, DBObject object) {
      super(KEYWORD, destName, object);
      Object value = object.get(keyword);
      if (!(value instanceof List) || ((List) value).size() != 2) {
        errorResult(coll, 16020, "the $ifNull operator requires an array of 2 operands");
      }
      @SuppressWarnings("unchecked") List<String> values = (List<String>) value;
      this.field = values.get(0);
      this.valueIfNull = values.get(1); // java.lang.Boolean cannot be cast to java.lang.String
  }
```

Is there any reason why ```valueIfNull``` cannot be treated as an ```java.lang.Object``` instead of an ```java.lang.String```?

--- a/src/main/java/com/github/fakemongo/impl/aggregation/Project.java
+++ b/src/main/java/com/github/fakemongo/impl/aggregation/Project.java
@@ -1,13 +1,5 @@
 package com.github.fakemongo.impl.aggregation;
 
-import com.github.fakemongo.impl.ExpressionParser;
-import com.github.fakemongo.impl.Util;
-import com.mongodb.BasicDBObject;
-import com.mongodb.DBCollection;
-import com.mongodb.DBObject;
-import com.mongodb.FongoDB;
-import com.mongodb.FongoDBCollection;
-import com.mongodb.MongoException;
 import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
 import java.util.Calendar;
@@ -17,9 +9,19 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.TimeZone;
+
 import org.bson.util.annotations.ThreadSafe;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+
+import com.github.fakemongo.impl.ExpressionParser;
+import com.github.fakemongo.impl.Util;
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBCollection;
+import com.mongodb.DBObject;
+import com.mongodb.FongoDB;
+import com.mongodb.FongoDBCollection;
+import com.mongodb.MongoException;
 
 /**
  * TODO : { project : { _id : 0} } must remove the _id field. If a $sort exist after...
@@ -219,7 +221,7 @@ public class Project extends PipelineKeyword {
     public static final String KEYWORD = "$ifNull";
 
     private final String field;
-    private final String valueIfNull;
+    private final Object valueIfNull;
 
     public ProjectedIfNull(String destName, DBCollection coll, DBObject object) {
       super(KEYWORD, destName, object);
@@ -235,7 +237,7 @@ public class Project extends PipelineKeyword {
     @Override
     void doWork(DBCollection coll, DBObject projectResult, Map<String, ProjectedAbstract> projectedFields, String key, Object value, String namespace) {
       createMapping(coll, projectResult, projectedFields, field, field, namespace, this);
-      createMapping(coll, projectResult, projectedFields, valueIfNull, valueIfNull, namespace, this);
+      createMapping(coll, projectResult, projectedFields, String.valueOf(valueIfNull), valueIfNull, namespace, this);
     }
 
     @Override

--- a/src/test/java/com/github/fakemongo/FongoAggregateProjectTest.java
+++ b/src/test/java/com/github/fakemongo/FongoAggregateProjectTest.java
@@ -1,5 +1,23 @@
 package com.github.fakemongo;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+
 import com.github.fakemongo.impl.Util;
 import com.github.fakemongo.junit.FongoRule;
 import com.mongodb.AggregationOutput;
@@ -7,21 +25,6 @@ import com.mongodb.BasicDBList;
 import com.mongodb.BasicDBObject;
 import com.mongodb.DBCollection;
 import com.mongodb.DBObject;
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.List;
-import java.util.Locale;
-import java.util.TimeZone;
-import org.assertj.core.api.Assertions;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-import org.junit.Ignore;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.ExpectedException;
-import org.junit.rules.RuleChain;
-import org.junit.rules.TestRule;
 
 public class FongoAggregateProjectTest {
 
@@ -417,6 +420,36 @@ public class FongoAggregateProjectTest {
         "                    { \"_id\" : 2, \"food\" : \"cherry\" },\n" +
         "                    { \"_id\" : 3, \"food\" : \"shepherd's\" },\n" +
         "                    { \"_id\" : 4, \"food\" : \"wasNull\" }]\n"), result);
+  }
+
+  /**
+   * See http://docs.mongodb.org/manual/reference/aggregation/ifNull/
+   */
+  @Test
+  public void testIfNullWithPrimitiveDataTypeValue() {
+	  DBCollection coll = fongoRule.newCollection();
+	  fongoRule.insertJSON(coll, "[{ _id: 1, item: { sec: \"dessert\", category: \"pie\", type: \"apple\" } },\n" +
+			  "{ _id: 2, item: { sec: \"dessert\", category: \"pie\", type: \"cherry\" } },\n" +
+			  "{ _id: 3, item: { sec: \"main\", category: \"pie\", type: \"shepherd's\" } },\n" +
+			  "{ _id: 4, item: { sec: \"main\", category: \"pie\" } }]");
+	  
+	  DBObject project = fongoRule.parseDBObject("{ $project: { food:\n" +
+			  "                                       { $ifNull: [ \"$item.type\",\n" +
+			  "                                                    true\n" +
+			  "                                                  ]\n" +
+			  "                                       }\n" +
+			  "                                }\n" +
+			  "                   }");
+	  
+	  AggregationOutput output = coll.aggregate(Arrays.asList(project));
+	  assertTrue(output.getCommandResult().ok());
+	  
+	  List<DBObject> result = (List<DBObject>) output.getCommandResult().get("result");
+	  assertNotNull(result);
+	  assertEquals(fongoRule.parse("[{ \"_id\" : 1, \"food\" : \"apple\" },\n" +
+			  "                    { \"_id\" : 2, \"food\" : \"cherry\" },\n" +
+			  "                    { \"_id\" : 3, \"food\" : \"shepherd's\" },\n" +
+			  "                    { \"_id\" : 4, \"food\" : true }]\n"), result);
   }
 
   /**

